src/apps/service/oauth.clj

```diff
@@ -69,15 +69,25 @@
 (defn authorization-uri
   "Generates an authorization URI for a remote API."
   [{:keys [api-name] :as server-info} username state-info]
-  (when-not (op/get-access-token api-name username)
```

Comment:
The problem was that `authorization-uri` was returning nil when the user already had an access token even if the access token wasn't any good. `authorization-uri` was called (here)[https://github.com/dennisroberts71/apps/blob/master/src/apps/service/apps.clj#L22-L25] to set the location header in the redirect.